### ISSUE TITLE
the `--shred --gui` command line is broken in new way

### ISSUE THREAD

**Issue by az0** [bleachbit/bleachbit#1124]
BleachBit has a simple integration with Windows Explorer that launches it with --no-uac --shred-gui. After merging 3a65231d469a68a8cb2fdc7f60a90fbdf005d63f, I tested on Windows 10 build 1829 (from ci.bleachbit.org). Before this commit, it was broken in this simple way described in https://github.com/bleachbit/bleachbit/issues/893 . Since the commit, it's still broken but more complex. The window initially shows one message. Then, it clears this message and shows another. Then, I think it goes in a loop.

Test command

```
del c:\temp\debug.log
echo deleteme>c:\temp\deleteme.txt
c:
cd \temp
"C:\Program Files (x86)\BleachBit\bleachbit_console.exe" --gui --no-uac --shred c:\temp\deleteme.txt --debug > c:\temp\debug.log 2>&1

```

Within a few seconds, the log is >3MB. Here are the first lines.

```
C:\Program Files (x86)\BleachBit\library.zip\win_unicode_console\__init__.py:31: RuntimeWarning: sys.stdin.encoding == 'utf-8', whereas sys.stdout.encoding == 'cp1252', readline hook consumer may assume they are the same
closest monitor (0) geometry = (0, 0)+(1366, 768), window geometry = (0, 0)+(1366, 728)
Traceback (most recent call last):
  File "bleachbit\Worker.pyo", line 260, in run
  File "bleachbit\Worker.pyo", line 336, in run_operations
KeyError: '_gui'
Traceback (most recent call last):
  File "bleachbit\Worker.pyo", line 260, in run
  File "bleachbit\Worker.pyo", line 336, in run_operations
KeyError: '_gui'
clean_operation('_gui'), options = '['files']'
ValueError in get_iter() when updating file size for tree path=0
ValueError in get_iter() when updating file size for tree path=0
elapsed time: 0 seconds
clean_operation('_gui'), options = '['files']'
ValueError in get_iter() when updating file size for tree path=0
ValueError in get_iter() when updating file size for tree path=0
elapsed time: 0 seconds
clean_operation('_gui'), options = '['files']'
[WinError 2] The system cannot find the file specified: 'c:\\temp\\deleteme.txt': Command to shred c:\temp\deleteme.txt
ValueError in get_iter() when updating file size for tree path=0
ValueError in get_iter() when updating file size for tree path=0
```

@rados  , you made a PR that changed the behavior.

@Marvin0401 , you worked on https://github.com/bleachbit/bleachbit/issues/893 and duplicate issue tickets.

**Comment by rados** [comment]
@az0 
I am working on the initial problem - the simple disappearing of the window without deleting the selected file when we execute 'Shred with Bleachbit' context menu command. It is related with #1080 also. I've found that it is a kind of regression introduced with 386afba0 and I decided to first write some tests around those changes and then to introduce a fix.

I'll trace why my fix for #844 (bdc04d0b) has added complexity to the issue. Thanks for pointing this out.

**Comment by rados** [comment]
@az0, I couldn't reproduce it. I've just checked out the merge commit (bdc04d0b), built the installer, installed from it and run your test command. My debug.log file is 3 KB and there are no duplications in it. Probably I am missing something...

**Comment by rados** [comment]
@az0, I've found something. With fixing #844 we started to show the 'confirm delete' dialog only when the respective property is checked and not showing it when the property is not checked. So, when we execute the context menu command and the property is not checked we don't show the dialog and this is why we are observing new behavior. So I suggest to check if this new behavior would still persists when the original problem is fixed.
